_posts/2020-05-20-hive-acid.md

```diff
@@ -0,0 +1,317 @@
+and write to Hive ACID tables via
+[Hive](https://cwiki.apache.org/confluence/display/Hive/Hive+Transactions)
+or [Spark](https://github.com/qubole/spark-acid) and use Presto to read these
+tables.
+
+# What are Hive ACID tables
+Hive ACID Tables, also called Transactional Tables in Hive, provide ACID
+semantics in Hive. This excerpt from
+[Hive documentation](https://cwiki.apache.org/confluence/display/Hive/Hive+Transactions)
+covers ACID traits well:
+> “ACID stands for four traits of database transactions:
+>Atomicity (an operation either succeeds completely or fails,
+>it does not leave partial data), Consistency (once an application performs an
+>operation the results of that operation are visible to it in every subsequent
+> operation), Isolation (an incomplete operation by one user does not cause
+>unexpected side effects for other users), and Durability (once an operation is
+>complete it will be preserved even in the face of machine or system failure).
+>These traits have long been expected of database systems as part of their
+>transaction functionality.“
+
+Several concepts like transactions, writeIds, deltas, locks, etc are added in
+Hive to achieve ACID semantics which we will cover later in the blog.
+
+# Need for Hive ACID Tables
+In any organisation, there is always a need to update or delete existing entries
+in tables e.g. a user writes or updates the review for the service availed a
+week back or a transaction status is changed after a day, etc.
+With regulations like GDPR/CCPA updates/deletes become even more frequent as the
+users can ask the organisation to delete the data on them and organisations are
+obligated to fulfill these requests.
+
+The standard practice to update data has been to overwrite the partition or
+table with the updated data but this is inefficient and unreliable. It takes a
+lot of resources to overwrite all of the existing data to update a few entries
+but more importantly there are issues around isolation when reads on old data
+are going on and the overwrite starts deleting that data. To solve these issues
+several solutions have been developed,many of them are covered
+[in this blog](https://www.qubole.com/blog/qubole-open-sources-multi-engine-support-for-updates-and-deletes-in-data-lakes/),
+and Hive ACID is one of them.
+
+# Concepts of Hive ACID tables
+
+## Types of Hive ACID tables
+There are two types of Hive ACID tables: Insert-Only and Full CRUD ACID tables.
+Following table compares the two:
+
+| Type of ACID Table     | Hive DML Operations Supported | Input Formats supported | Synthetic columns in file? | Additional Table Properties   |
+|:---------------------:|:--------------------------:|:-------------------------:|:----------------------------:|:-----------------------------:|
+|Insert Only ACID Tables| INSERT                     | All input formats         |    No                        | 'transactional'='true', 'transactional_properties'='insert_only' |
```

Comment:
Add code quotes (backticks) around the table properties so that they're rendered as code (also prevents the quotes from being made "pretty" in the HTML)